src/main/java/org/jabref/logic/database/DuplicateCheck.java

```diff
@@ -303,6 +308,19 @@ public boolean isDuplicate(final BibEntry one, final BibEntry two, final BibData
             return true;
         }
 
+        // check DOI
+        Optional<DOI> oneDOI = one.getDOI();
+        Optional<DOI> twoDOI = two.getDOI();
+        if (oneDOI.isPresent() && twoDOI.isPresent()) {
+            return oneDOI.get().getDOI().equalsIgnoreCase(twoDOI.get().getDOI());
```

Comment:
I think, using `org.jabref.model.entry.identifier.DOI#equals` makes more sense.

Furthermore, [Objects.equals](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-) should be used

```suggestion
            return Objects.equals(oneDOI, twoDOI);
```